src/libcore/ptr.rs

```diff
@@ -228,6 +228,55 @@ pub const fn null<T>() -> *const T { 0 as *const T }
 #[rustc_promotable]
 pub const fn null_mut<T>() -> *mut T { 0 as *mut T }
 
+#[repr(C)]
+pub(crate) union Repr<T> {
+    pub(crate) rust: *const [T],
+    rust_mut: *mut [T],
+    pub(crate) raw: FatPtr<T>,
+}
+
+#[repr(C)]
+pub(crate) struct FatPtr<T> {
+    data: *const T,
+    pub(crate) len: usize,
+}
+
+/// Forms a slice from a pointer and a length.
+///
+/// The `len` argument is the number of **elements**, not the number of bytes.
+/// Note: using a `len` value larger than `isize::max_value()` is not recommended, as dereferencing
```

Comment:
`len` can be larger than `isize::max_value()` if `T` is a ZST. It's really `len * size_of::<T>()` that shouldn't be larger than `isize::max_value()`.